GiniVision/Classes/QRCodeDetectedPopupView.swift

```diff
@@ -0,0 +1,170 @@
+                          constant: padding.left)
+        Contraints.active(item: qrImage, attr: .top, relatedBy: .equal, to: self, attr: .top, constant: padding.top)
+        Contraints.active(item: qrImage, attr: .bottom, relatedBy: .equal, to: self, attr: .bottom,
+                          constant: -padding.bottom)
+        Contraints.active(item: qrImage, attr: .trailing, relatedBy: .equal, to: qrText, attr: .leading,
+                          constant: -padding.right / 2, priority: 999)
+        
+        Contraints.active(item: qrText, attr: .centerY, relatedBy: .equal, to: qrImage, attr: .centerY)
+        Contraints.active(item: qrText, attr: .trailing, relatedBy: .lessThanOrEqual, to: proceedButton, attr: .leading,
+                          constant: -padding.right)
+        Contraints.active(item: proceedButton, attr: .centerY, relatedBy: .equal, to: qrImage, attr: .centerY)
+        Contraints.active(item: proceedButton, attr: .trailing, relatedBy: .equal, to: self, attr: .trailing,
+                          constant: -padding.right)
+        if let minButtonWidth = proceedButton.titleLabel?.intrinsicContentSize.width {
+            Contraints.active(item: proceedButton, attr: .width, relatedBy: .equal, to: nil, attr: .notAnAttribute,
+                              constant: minButtonWidth)
+        }
+
+    }
+    
+    fileprivate func addShadow() {
+        self.layer.shadowOffset = CGSize(width: 0, height: 2)
+        self.layer.shadowRadius = 0.8
+        self.layer.shadowOpacity = 0.2
+        self.layer.shadowColor = UIColor.black.cgColor
+    }
+    
+    @objc fileprivate func didTapDoneAction() {
+        didTapDone()
+    }
+}
+
+// MARK: - Animations
+
+extension QRCodeDetectedPopupView {
+    func show(after seconds: TimeInterval = 0) {
+        DispatchQueue.main.asyncAfter(deadline: .now() + seconds, execute: { [weak self] in
+            guard let `self` = self, let superview = self.superview else { return }
+            self.bottomConstraint?.constant = -self.margin.bottom
+            UIView.animate(withDuration: 0.5,
+                           delay: 0,
+                           options: [.curveEaseInOut],
+                           animations: {
+                            superview.layoutIfNeeded()
+            }, completion: { [weak self] _ in
+                guard let `self` = self else { return }
+                self.hide(after: self.hiddingDelay)
+            })
+        })
+    }
+    
+    func hide(after seconds: TimeInterval = 0, completion: (() -> Void)? = nil) {
+        DispatchQueue.main.asyncAfter(deadline: .now() + seconds, execute: { [weak self] in
+            guard let `self` = self, let superview = self.superview else { return }
+            self.bottomConstraint?.constant = self.imageSize.height +
+                self.padding.top +
+                self.padding.bottom +
+                self.margin.bottom
+            UIView.animate(withDuration: 0.5,
```

Comment:
This 0.5 is duplicated with the show method. Also, I keep recommending to have animation durations set as parameters (slow, medium, fast).